src/framework/core/js/Fluid.js

```diff
@@ -2851,32 +2851,88 @@ var fluid = fluid || fluid_3_0_0;
 
     // Message resolution and templating
 
-   /**
-    * Converts a string to a regexp with the specified flags given in parameters
-    * @param {String} a string that has to be turned into a regular expression
-    * @param {String} the flags to provide to the reg exp
-    */
-    // TODO: this is an abominably inefficient technique for something that could simply be done by means of indexOf and slice
-    fluid.stringToRegExp = function (str, flags) {
-        return new RegExp(str.replace(/[\-\[\]{}()*+?.,\\\^$|#\s]/g, "\\$&"), flags);
+    /**
+     *
+     * Take an original object and represent it using top-level sub-elements whose keys are EL Paths.  For example,
+     * `originalObject` might look like:
+     *
+     * ```
+     * {
+     *   deep: {
+     *     path: {
+     *       value: "foo",
+     *       emptyObject: {},
+     *       array: [ "peas", "porridge", "hot"]
+     *     }
+     *   }
+     * }
+     * ```
+     *
+     * Calling `fluid.flattenObjectKeys` on this would result in a new object that looks like:
+     *
+     * ```
+     * {
+     *   "deep": "[object Object]",
+     *   "deep.path": "[object Object]",
+     *   "deep.path.value": "foo",
+     *   "deep.path.array": "peas,porridge,hot",
+     *   "deep.path.array.0": "peas",
+     *   "deep.path.array.1": "porridge",
+     *   "deep.path.array.2": "hot"
+     * }
+     * ```
+     *
+     * This function preserves the previous functionality of displaying an entire object using its `toString` function,
+     * which is why many of the paths above resolve to "[object Object]".
+     *
+     * This function is an unsupported non-API function that is used in by `fluid.stringTemplate` (see below).
+     *
+     * @param `originalObject` `{Object}` - An object.
+     * @returns `{Object}` - A representation of the original object that only contains top-level sub-elements whose keys are EL Paths.
```

Comment:
We should try to be consistent with these doc comments in case anyone does write a parser for them. i) I think the general convention is for @return rather than @returns, and ii) it doesn't look like we escape these with backticks

Reply:
I guess I got the idea that it was OK by omission, i.e. comments [like this one](https://github.com/GPII/gpii-diff/pull/1#pullrequestreview-75075782) that overlooked the backticks.  It's a shame, as I finally trained myself to use them consistently, but I agree, prior art in Infusion doesn't use it (or `returns`).  Will fix shortly.

Reply:
In reflecting on it further, I think I picked up the habit when adapting JSDocs into markdown.

Reply:
Agree on the goal and the specifics.  I notice a few other inconsistencies within the package, for example, some JSDocs use `@param variableName {Type}`, others use `@param {Type} variableName`.  I propose standardising on the former.

Reply:
Yes, that seems good - feel free to normalise any examples of the other that you find

Reply:
Looks like we can at least enforce the consistent use of `@returns`:  https://eslint.org/docs/rules/valid-jsdoc#options

If you're game, I can submit a PR against our comment ESLint rules to add that.

Reply:
Anyway, fixed throughout.